### PR TITLE
[CLOUDGA-23989] Use encoded db credentials

### PR DESCRIPTION
**Pull request by bansal01yash**
## Summary
Changed terraform cluster creation to use b64 encoded db credentials

## Test Plan
Tested by creating a cluster and connecting using given credentials

### CODE REVIEW

File: managed/resource_cluster.go [hunk 1]
```diff
@@ -1018,16 +1024,17 @@ func (r resourceCluster) Create(ctx context.Context, req tfsdk.CreateResourceReq
...
+		encryptedCredentials.SetYcql(*openapiclient.NewEncryptedDBCredentials(b64(plan.Credentials.YCQLUsername.Value), b64(plan.Credentials.YCQLPassword.Value)))
 	} else {
-		credentials.SetYsql(*openapiclient.NewDBCredentials(plan.Credentials.Username.Value, plan.Credentials.Password.Value))
-		credentials.SetYcql(*openapiclient.NewDBCredentials(plan.Credentials.Username.Value, plan.Credentials.Password.Value))
+		encryptedCredentials.SetYsql(*openapiclient.NewEncryptedDBCredentials(b64(plan.Credentials.Username.Value), b64(plan.Credentials.Password.Value)))
+		encryptedCredentials.SetYcql(*openapiclient.NewEncryptedDBCredentials(b64(plan.Credentials.Username.Value), b64(plan.Credentials.Password.Value)))
 	}
-
-	createClusterRequest := *openapiclient.NewCreateClusterRequest(*clusterSpec, *credentials)
+	createClusterRequest.SetEncryptedDbCredentials(*encryptedCredentials)
```
Comment (harshitha-yb) [inline managed/resource_cluster.go:1037]: not sure but we might have an issue with this when we mark encrypted_db_credentials as required? Let's keep it as `createClusterRequest := *openapiclient.NewCreateClusterRequest(*clusterSpec, *encryptedCredentials)` only

Comment (bansal01yash) [reply managed/resource_cluster.go:1037]: You can only pass required params there, optionals have to be added separately. 
```
managed/resource_cluster.go:1036:78: too many arguments in call to openapiclient.NewCreateClusterRequest
	have (openapi.ClusterSpec, openapi.CreateClusterRequestEncryptedDbCredentials)
	want (openapi.ClusterSpec)
make: *** [build] Error 1```

Comment (harshitha-yb) [reply managed/resource_cluster.go:1037]: Cool then will have to make the change later

**Review by harshitha-yb — APPROVED** [review]

**Review by daniel-yb — COMMENTED** [review]
@posriniv please review

**Review by posriniv — APPROVED** [review]
Looks good to me!

**Pull request merged**